mlc_llm/utils.py

```diff
@@ -81,11 +81,13 @@ def argparse_postproc_common(args: argparse.Namespace) -> None:
         "minigpt": "minigpt",
     }
     try:
-        with open(os.path.join(args.model_path, "config.json"), encoding="utf-8") as i_f:
+        with open(
+            os.path.join(args.model_path, "config.json"), encoding="utf-8"
+        ) as i_f:
             config = json.load(i_f)
             args.model_category = config["model_type"]
-    except IOError as error:
-        pass
```

Comment:
When `config.json` does not exist, the exception will be `AttributeError` instead of `IOError`. Change to `Exception` to capture all kinds of exceptions.

Reply:
cc @sbelcmu

Reply:
Got it thanks. On a side note, do we want `args.model_category = ""`? I think this might silently fail, and it would be better to have `args.model_category = None` so that the assert catches it and it would not silently fail.

Reply:
We have some testing scripts like https://github.com/mlc-ai/mlc-llm/blob/main/tests/evaluate.py and https://github.com/mlc-ai/mlc-llm/blob/main/tests/debug/compare_lib.py, which focus on runtime, also leverage `argparse_postproc_common` but do not have args like `args.model_path`. So for their cases they also don’t have `args.model_category`, and using the assertion below might make those scripts unable to run.

We might need to further refine both here and the testing scripts a bit to make the logic more elegant in the future. But not a huge blocker for now.